app/models/plan.rb

```diff
@@ -0,0 +1,24 @@
+class Plan < ApplicationRecord
+  validates :title, length: { in: 2..100 }
+  validates :since_date, presence: true
+  validates :to_date, presence: true
+  validates :goal, presence: true, numericality: { other_than: 0 }
+
+  def status
+    return @result if @result
+
+    products = Product.where(user_id: user_id).order('created_at')
+    products = products.where('created_at > ?', since_date).where('created_at < ?', to_date)
```

Comment:
Kiedy stworzyłem plan na ten tydzień (10-16 grudnia) to wydatki dodane w niedzielę nie pokazują się w planie.

Proszę spróbować tego - https://apidock.com/rails/DateTime/end_of_day
```
products = products.where(('created_at > ?', since_date.beginning_of_day).where('created_at < ?', to_date.end_of_day)
```